presto-main/src/main/java/com/facebook/presto/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2182,6 +2194,7 @@ private PhysicalOperation createLookupJoin(JoinNode node,
             boolean isBroadcastJoin = distributionType.isPresent() && distributionType.get() == REPLICATED;
 
             LocalExecutionPlanContext buildContext = context.createSubContext();
+            buildContext.setHasHashBuilder(true);
```

Comment:
At this step,  `driverInstanceCount` is empty.

Reply:
There is an issue with visitExchange, where if have a gather or sorting exchange, it requires one task and so tries to set the driverInstanceCount to one.  That means that you need to do a check here for what exchange type you'll end up with further down so you don't break that part of the code.  I added a check, but it felt very hacky and fragile because there could be some other circumstance in the future that would cause similar issues that we hadn't checked for.  That's why I switched to this approach.

Reply:
Make sense if `driverInstanceCount` should always take high priority.